### PR TITLE
CORE-6342: Add role to static registration context

### PR DESCRIPTION
**Pull request by yift-r3**
Adds optional notary details and roles to the static registration context.

 Runtime PR in https://github.com/corda/corda-runtime-os/pull/2264

### CODE REVIEW

File: data/membership-schema/src/main/resources/net/corda/schema/membership/member/static/registration/1.0/corda.member.static.registration.json [hunk 1]
```diff
@@ -4,13 +4,39 @@
   "title": "Corda static member registration context schema",
   "description": "Definition of the registration context required for registering a member in a static group.",
   "type": "object",
+  "patternProperties": {
+    "^corda.roles.[0-9]+$": {
+      "description": "The member role. Can be multiple specified by index.",
+      "type": "string",
+      "enum": ["notary", "member"],
```
Comment (charlieR3) [inline data/membership-schema/src/main/resources/net/corda/schema/membership/member/static/registration/1.0/corda.member.static.registration.json]: What is the 'member' role in this case?

Comment (yift-r3) [reply data/membership-schema/src/main/resources/net/corda/schema/membership/member/static/registration/1.0/corda.member.static.registration.json]: Don't we need a role for a "Vanilla" member?

Comment (charlieR3) [reply data/membership-schema/src/main/resources/net/corda/schema/membership/member/static/registration/1.0/corda.member.static.registration.json]: I was thinking no role is a standard member. Then adding a role give additional capabilities. A notary member should be able to do everything a regular member can do with some additional functionality.

**Review by charlieR3 — APPROVED** [review]

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — APPROVED** [review]
LGTM

**Pull request merged**